distributed/tests/test_semaphore.py

```diff
@@ -140,7 +140,11 @@ async def test_async_ctx(s, a, b):
 
 @pytest.mark.slow
 def test_worker_dies():
-    with cluster() as (scheduler, workers):
+    with cluster(
+        config={
+            "distributed.scheduler.locks.lease-timeout": "0.1s",
+        }
+    ) as (scheduler, workers):
```

Comment:
I realize this predates this PR, but out of curiosity is there a reason to not use [`gen_cluster`](https://github.com/dask/distributed/blob/de6db936e0cbfcd7d71c4258fd72b11994ec37ed/distributed/utils_test.py#L829) here instead of the `cluster` context manager? Maybe it's something to do with `gen_cluster` and the `pytest.mark.slow` marker not working well together? 

I'm wondering because `gen_cluster` is used more broadly throughout the codebase and already has a `config=` keyword argument.

Reply:
Most (all?) APIs we can use for the async setup is a somehow graceful shutdown. In this case, we're actually terminating the process, removing any possibility for a final release or graceful disconnect. Closest thing to a real VM dying. 

> Maybe it's something to do with gen_cluster and the pytest.mark.slow marker not working well together?

I'm not aware of anything like this, it's just the async/sync thing